core/kotlinx-coroutines-core/src/main/kotlin/kotlinx/coroutines/experimental/channels/Subscribables.kt

```diff
@@ -0,0 +1,36 @@
+package kotlinx.coroutines.experimental.channels
+
+import kotlin.properties.ReadOnlyProperty
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
+
+
+public interface Subscribable<out T> {
+  public fun openSubscription(): SubscriptionReceiveChannel<T>
+}
+
+public interface SubscribableValue<out T> : Subscribable<T>, ReadOnlyProperty<Any, T> {
+
+  /**
+   * Current value if any.
+   *
+   * Access this property may throws [IllegalStateException] if there is no value set yet.
+   *
+   * Use [valueOrNull] to get `null` when the value is not set or `openSubscription().consume { receive() }` to suspend until a value is available.
+   */
+  public val value: T
+
+  public val valueOrNull: T?
+
+  override fun getValue(thisRef: Any, property: KProperty<*>): T = value
+}
+
+public interface SubscribableVariable<T> : SubscribableValue<T>, ReadWriteProperty<Any, T> {
+  public override var value: T
+
+  override fun getValue(thisRef: Any, property: KProperty<*>): T = value
```

Comment:
Is it already defined in `SubscribableValue`?

Reply:
Yes, but Kotlin compiler enforce to redeclare it, because it is inherited from two interfaces (`SubscribableValue` and `ReadWriteProperty`)

Reply:
https://kotlinlang.org/docs/reference/classes.html#overriding-rules

Reply:
Ok